### ISSUE TITLE
Move Content - Content File locked

### ISSUE THREAD

**Issue by Zodiaksl** [CryoByte33/steam-deck-utilities#46]
Hello,

I've just gone through your updated tweaks with GUI. I've watched the related video and I've applied all of them with the recommended settings, with 16gb swap and 4gb VRAM and every other option.

It's been about a day since and I'm trying to reorganize some of my games, shuffling them from the SD card to my internal 2TB SSD and I'm now getting the following error message while trying to move things in Gaming mode.

Move Content
Moving content for the following apps has failed: 
- Rayman Legends : content file locked

Has anyone else experienced this? Please let me know.

Thank you for your continued efforts in improving performance.

**Comment by jwderoos** [comment]
This is an issue with steam in general. There are some workarounds available, but they do not always work. https://www.reddit.com/r/SteamDeck/comments/10880pf/content_file_locked_workaround/

**Comment by samuelbak** [comment]
I have the same issue, after I used the sync shadercache content I can't move any games. I don't think this is a "steam general issue" seems that it's caused by CryoTools

**Comment by CryoByte33** [comment]
There's no reason that this should be caused since I'm just using the OS' copy handler. Does the issue go away if Steam is closed? It sounds Like Steam is holding on to the file handle.

**Comment by Zodiaksl** [comment]
I was in Gaming mode and tried moving games in Settings > Storage, trying to move games from the SD card to the Internal storage.

I had done the Sync and Shader thing before, then rebooted into gaming mode and I get that error.

**Comment by Zodiaksl** [comment]
Can confirm the error for the following games:

Bloodstained: Curse of the Moon (20.75MB)
Bloodstained: Curse of the Moon 2 (69.42MB)
Downwell (43.09MB)
Momodora: Reverie Under the Moonlight (76.84MB)

And the list goes on.

Trying to move from my SD Card to my Internal storage. So it's not a single instance.

SD Card: Sandisk Extreme 1TB
Internal: WD SN740 2TB

**Comment by Zodiaksl** [comment]
Moving games in Gaming mode under Settings > Storage, FROM my Internal Storage TO the SD card has not resulted in an error so far (fingers crossed).

I've tested those a few times.

After doing the Sync and moving games from Internal to SD Card and back, do we have to resync each time?

**Comment by CryoByte33** [comment]
> Trying to move from my SD Card to my Internal storage. So it's not a single instance.

I'm still unable to replicate with either previously-installed or newly-installed games. I'm not sure what the difference between our setups is.
* Which SteamOS version are you running?
* If you uninstall a game, reboot, reinstall, launch, and then try to sync the data, does this still happen?
* Have you previously used any other shader-moving tools?
* What is the output of `stat` on an affected shader directory?

> After doing the Sync and moving games from Internal to SD Card and back, do we have to resync each time?

You only have to run sync when a game has changed location. If it's gone to SD and back prior to running sync, then there is no need as the shaders never moved. However, if you move it to SD and then sync, you'll need to resync if you move it back to internal storage.

**Comment by Zodiaksl** [comment]
Channel: Beta

OS Version: 3.4.4
OS Build: 20221228:1
Kernel Version 5.13.0-valve36-1-neptune

Steam
Steam Version: 1676935611
Steam Web Build Date: Mon, Feb 20 3:39 PM UTC -05:00
Steam API Version: SteamClient020

**Comment by CryoByte33** [comment]
> This is an issue with steam in general. There are some workarounds available, but they do not always work. https://www.reddit.com/r/SteamDeck/comments/10880pf/content_file_locked_workaround/

I just got around to reading this link. It sounds like there's a problem while copying some files for some games, but not moving for some reason. This is problematic since I don't want to accidentally delete files or leave something half-moved in the event of a failure, the copy is there for safety.

Based on the nature of the failure, I'm curious if the difference is microSD cards, with some having read/write lock support and others not. 

Does anyone affected have another microSD to try? I only have the one I can't replicate with.

**Comment by Zodiaksl** [comment]
Hello. Unfortunately, I only have more of the same card as the Sandisk Extreme 1TB is one of the most highly recommended :(

**Comment by CryoByte33** [comment]
I only have a 256GB Samsung Evo, which is definitely not the top tier of card. I wonder if _my_ card doesn't support locking, thus why I'm not getting an issue?

Barring anyone else being able to help here, I'll do my best to get a good card soon.

**Comment by Zodiaksl** [comment]
No worries Bud, you do excellent work.

Not worried about this in the slightest since I can just redownload games. Just wanting to help you solve a problem is all :)

**Comment by CryoByte33** [comment]
Just to be clear, I want to fix it too, just workshopping ideas 😅

**Comment by CryoByte33** [comment]
Just got another report of this issue from Discord, with the exact same microSD card, a 1TB Sandisk Extreme. 

If anyone out there has a Sandisk Extreme and wants to test if they have this issue, then that would be a big help.

**Comment by SeimusS** [comment]
As mentioned on Discord, I was able to reproduce the issue on Samsung SD as well.

I tried:
1. Deck into sleep and as well 
2. Rebooting it

Both actually worked and I was able to move the game within the Game mode.

**Comment by CryoByte33** [comment]
Yeah, this seems to be less related to CU and more related to how Steam holds files open. As @SeimusS mentioned, forcing some sort of sleep process lets Steam release the file handle.

I'm not sure if there's anything we can do from our side aside from murdering the Steam process and starting it again, which I don't think is a great idea.
